### PR TITLE
chore(refactor): get rid of circular references and enable ESLint rule

### PR DESCRIPTION
**Pull request by keraf**
This enables the [import/no-cycle](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-cycle.md) rule in ESLint and fixes all reported issues.

### CODE REVIEW

File: packages/suite/src/components/suite/FiatValue/index.tsx
Comment (slowbackspace) [inline packages/suite/src/components/suite/FiatValue/index.tsx]: it is not a currency, it stores fiat value of amount in given targetCurrency. fiat was slightly less confusing since there is variable targetCurrency that is passed to `currency` prop few lines below :)

Comment (slowbackspace) [reply packages/suite/src/components/suite/FiatValue/index.tsx]: what about fiatValue/fiatAmount? :)

File: packages/suite/src/views/dashboard/components/PortfolioCard/components/DashboardGraph/index.tsx
Comment (slowbackspace) [inline packages/suite/src/views/dashboard/components/PortfolioCard/components/DashboardGraph/index.tsx:46]: dropped React.memo is intentional?

Comment (slowbackspace) [reply packages/suite/src/views/dashboard/components/PortfolioCard/components/DashboardGraph/index.tsx:46]: oh found it

File: packages/suite/src/components/suite/FiatValue/index.tsx
Comment (keraf) [reply packages/suite/src/components/suite/FiatValue/index.tsx:83]: `fiat` and `fiatCurrency` were already taken, couldn't think of anything better :smile:

File: packages/suite/src/views/dashboard/components/PortfolioCard/components/DashboardGraph/index.tsx
Comment (keraf) [reply packages/suite/src/views/dashboard/components/PortfolioCard/components/DashboardGraph/index.tsx:46]: I moved it at the bottom (export).

File: packages/suite/src/components/suite/FiatValue/index.tsx
Comment (keraf) [reply packages/suite/src/components/suite/FiatValue/index.tsx]: b5ba4db

File: packages/blockchain-link/src/types/blockbook/common.ts
Comment (szymonlesisz) [inline packages/blockchain-link/src/types/blockbook/common.ts]: i have another idea for that:
https://github.com/trezor/trezor-suite/pull/3474/commits/49dd3ec68dcde277730480877de06756ac226964

Comment (keraf) [reply packages/blockchain-link/src/types/blockbook/common.ts]: Rebased on develop and reverted my changes in blockchain-link.

**Review by slowbackspace — APPROVED** [review]
lgtm, but let's wait for the freeze before we merge this, just to be safe 🤞

**Pull request merged**
